src/Text/Parsing/StringParser/String.purs

```diff
@@ -111,3 +115,39 @@ anyLetter = lowerCaseChar <|> upperCaseChar <?> "Expected a letter"
 -- | Match a letter or a number.
 alphaNum :: Parser Char
 alphaNum = anyLetter <|> anyDigit <?> "Expected a letter or a number"
+
+-- | Build the regular expression from the pattern and match it, ensuring
+-- | that the pattern only attempts to match from the start of the target.
+regex' :: String -> Parser String
+regex' pat =
+    case Regex.regex pattern noFlags of
+      Left _ ->
+        fail $ "Text.Parsing.StringParser.String.regex': illegal regex " <> pat
+      Right r ->
+        regex r
+    where
+      pattern =
+        case stripPrefix (Pattern "^") pat of
+          Nothing ->
+            "^" <> pat
+          _ ->
+            pat
+
+-- | Match the regular expression.
+regex :: Regex.Regex -> Parser String
+regex r =
+  Parser \{ str, pos } ->
+    let
+      remainder = drop pos str
+    in
+      -- reduce the possible array of matches to 0 or 1 elements to aid Array pattern matching
+      case uncons $ fromMaybe [] $ Regex.match r remainder of
+        Just { head: Just matched, tail: _ }  ->
+          -- only accept matches at position 0
+          case stripPrefix (Pattern matched) remainder of
```

Comment:
Hmm - you're right - I hadn't thought of that either.  I can't think of a good way of getting position information for each match.  I suppose, if we want to retain precompilation, there is the alternative of using a wrapper for Regex, hiding the default constructor and then providing a constructor that prefixes the '^' character.  So we'd have something like:

```purescript
data ParserRegex = ParserRegex Regex.Regex

-- | Construct a ParserRegex ensuring patterns are always prefaced with '^'.
parserRegex :: String -> Either String ParserRegex

-- | Build the regular expression from the pattern and match it
-- | (fail on patterns in error).
regex' :: String -> Parser String

-- | Match the regular expression.
regex :: ParserRegex -> Parser String
```
This doesn't introduce any further dependencies, but it is a little cumbersome.  Otherwise, I'd be very happy to revert to my original version.